src/domain/profile/modal/__tests__/EditProfileModal.test.tsx

```diff
@@ -88,23 +89,31 @@ it('renders snapshot correctly', () => {
 it('should allow all fields to be filled', async () => {
   initModal();
   const result = getWrapper();
-  const { getByLabelText, getAllByLabelText, queryByDisplayValue } = result;
+  const { getByRole, getAllByRole, queryByDisplayValue } = result;
 
-  fireEvent.change(getByLabelText('Sähköpostiosoite *'), {
-    target: { value: formData.email },
-  });
-  fireEvent.change(getByLabelText('Puhelinnumero *'), {
+  fireEvent.change(
+    getByRole('textbox', {
+      name: 'Sähköpostiosoite *',
+    }),
+    {
+      target: { value: formData.email },
+    }
+  );
+  fireEvent.change(getByRole('textbox', { name: 'Puhelinnumero *' }), {
     target: { value: formData.phoneNumber },
   });
-  fireEvent.change(getByLabelText('Etunimi *'), {
+  fireEvent.change(getByRole('textbox', { name: 'Etunimi *' }), {
```

Comment:
Should we use testid for these to have more robust testing ids in english?
https://testing-library.com/docs/queries/bytestid/

Reply:
There is a good reason to use those roles: they tests the accessibility names so they tests how the screen reader reads the fields!

Reply:
so, if you change the test language to English, then you need to use English texts too.